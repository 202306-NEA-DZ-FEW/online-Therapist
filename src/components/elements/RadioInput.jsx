const RadioInput = ({ onChange, name, value, text, width }) => {
    return (
        <div>
            <label className='cursor-pointer'>
                <input
                    type='radio'
                    className='peer sr-only'
                    name={name}
                    value={value}
                    onChange={onChange}
                />
                <div
                    className={`lg:w-${width} max-w-xl rounded-md bg-SkyBlue p-5  hover:bg-LightTeal hover:text-black hover:ring-Teal peer-checked:bg-Teal peer-checked:text-white text-Gray ring-2 ring-transparent transition-all hover:shadow peer-checked:ring-blue-400 peer-checked:ring-offset-2`}
                >
                    <div className=' gap-1'>
                        <div className='flex items-center justify-between'>
                            <p className='lg:text-xl font-atkinson uppercase'>
                                {text}
                            </p>
                        </div>
                    </div>
                </div>
            </label>
        </div>
    );
};

export default RadioInput;
