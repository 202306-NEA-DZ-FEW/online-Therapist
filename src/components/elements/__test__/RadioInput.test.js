import renderer from "react-test-renderer";

import RadioInput from "../RadioInput";

it("renders correctly", () => {
    const tree = renderer.create(<RadioInput />).toJSON();
    expect(tree).toMatchSnapshot();
});
