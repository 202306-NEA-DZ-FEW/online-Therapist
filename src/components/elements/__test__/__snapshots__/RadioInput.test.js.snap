// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <label
    className="cursor-pointer"
  >
    <input
      className="peer sr-only"
      type="radio"
    />
    <div
      className="lg:w-undefined max-w-xl rounded-md bg-SkyBlue p-5  hover:bg-LightTeal hover:text-black hover:ring-Teal peer-checked:bg-Teal peer-checked:text-white text-Gray ring-2 ring-transparent transition-all hover:shadow peer-checked:ring-blue-400 peer-checked:ring-offset-2"
    >
      <div
        className=" gap-1"
      >
        <div
          className="flex items-center justify-between"
        >
          <p
            className="lg:text-xl font-atkinson uppercase"
          />
        </div>
      </div>
    </div>
  </label>
</div>
`;
